src/QSymbolicsBase/basic_ops_homogeneous.jl

```diff
@@ -81,8 +81,8 @@ julia> k₁ + k₂
     _arguments_precomputed
 end
 function SAdd{S}(d) where S 
-    xs = [c*obj for (c,obj) in d]
-    length(d)==1 ? first(xs) : SAdd{S}(d,Set(xs),xs)
+    terms = flattenop(+,[c*obj for (c,obj) in d])
+    length(d)==1 ? first(terms) : SAdd{S}(d,Set(terms),terms)
```

Comment:
This looks a bit suspicious. There is no one-to-one correspondence between the dictionary, set, and list forms anymore. I think what you want is actually `countmap_flatten` as seen here https://github.com/QuantumSavory/QuantumSymbolics.jl/blob/079ea5e6f3c14b34a791aef99704035f1841fb29/src/QSymbolicsBase/basic_ops_homogeneous.jl#L97

Given that there is already flattening done in the `+` operation, I think we can just not do any flattening here. Just take the dictionary and do not worry about processing it. Any normal use of `SAdd`, usually through the use of `+` will already take care of flattening. This is a fairly general statement: fancy processing should be done in calls to `+` and `*`, not in constructors. 

Also, something I did not notice previously, the variable names are misleading here. It should be `(obj, coef) in d` not `(coef, obj) in d`.

Reply:
For this particular case, unlike the other constructors, we probably still want to keep the inner constructor that makes sure to create the set and vector.